canonical-web-and-design/maas-ui
# Use useParams from react-router.

## Done

- As of [react-router 5.1.0](https://github.com/ReactTraining/react-router/releases/tag/v5.1.0), some new hooks are supported (`useParams`, `useLocation`, `useRouteMatch`). This branch removes our custom `useParams` hook and uses the new one provided by `react-router`. 

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Ensure views that use `useParams` still work as expected. (KVM details, edit API key, edit User etc.)
